hack/generator/pkg/codegen/pipeline/implement_convertible_status_interface.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/conversions"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/functions"
+)
+
+// ImplementConvertibleStatusInterfaceStageId is the unique identifier for this pipeline stage
+const ImplementConvertibleStatusInterfaceStageId = "implementConvertibleStatusInterface"
+
+func ImplementConvertibleStatusInterface(idFactory astmodel.IdentifierFactory) Stage {
```

Comment:
The code of this and the Spec stage are basically the same, with just the word "spec" replaced with "status". Can this just be one stage factory that gets passed in some kind of strategy that encapsulates which of spec or status should be used?

Reply:
I tried this refactoring, but the code for the stage factory was pretty hard to read; I think it's better to leave them as separate (but similar) stages.

Reply:
Fair enough!